calcOutdatedMetrics.js

```diff
@@ -1,69 +1,80 @@
+const { exec } = require("child_process");
+
 process.stdin.resume();
-process.stdin.setEncoding('utf8');
-let data = '';
+process.stdin.setEncoding("utf8");
+let data = "";
 
-process.stdin.on('data', function (chunk) {
+process.stdin.on("data", function (chunk) {
   data += chunk;
 });
 
-process.stdin.on('end', function () {
-  const jsonLines = data.split('\n');
-  const lineWithDependencyTable = jsonLines.find(json =>
-    json.includes('{"type":"table"'),
-  );
+process.stdin.on("end", function () {
+  exec(
+    "cat package.json | jq '.dependencies, .devDependencies | keys | length' | paste -s -d+ - | bc",
+    (err, stdout) => {
+      const jsonLines = data.split("\n");
+      const lineWithDependencyTable = jsonLines.find((json) =>
+        json.includes('{"type":"table"')
+      );
 
-  if (!lineWithDependencyTable) {
-    console.log('data:', data);
-    throw new Error(
-      `Did not receive output from 'yarn outdated'. data: ${data}`,
-    );
-  }
-  const parsedData = JSON.parse(lineWithDependencyTable);
-  const packageList = parsedData.data.body;
+      if (!lineWithDependencyTable) {
+        console.log("data:", data);
+        throw new Error(
+          `Did not receive output from 'yarn outdated'. data: ${data}`
+        );
+      }
+      const parsedData = JSON.parse(lineWithDependencyTable);
+      const packageList = parsedData.data.body;
 
-  const numberOfPackages = packageList?.length;
+      const numberOfPackages = parseInt(stdout);
 
-  const upToDate = [];
-  const behindPatch = [];
-  const behindMinor = [];
-  const behindMajor = [];
+      const behindPatch = [];
+      const behindMinor = [];
+      const behindMajor = [];
 
-  packageList?.forEach(package => {
-    const [name, currentVersion, wanted, latestVersion] = package;
-    const [currentMajor, currentMinor, currentPatch] = currentVersion;
-    const [latestMajor, latestMinor, latestPatch] = latestVersion;
+      packageList?.forEach((package) => {
+        const [name, currentVersion, wanted, latestVersion] = package;
+        const [currentMajor, currentMinor, currentPatch] = currentVersion
+          .split(".")
+          .map(parseInt);
+        const [latestMajor, latestMinor, latestPatch] = latestVersion
+          .split(".")
+          .map(parseInt);
 
-    if (currentMajor >= latestMajor) {
-      if (currentMinor >= latestMinor) {
-        if (currentPatch >= latestPatch) {
-          upToDate.push(package);
+        if (currentMajor >= latestMajor) {
+          if (currentMinor >= latestMinor) {
+            if (currentPatch >= latestPatch) {
+              throw new Error(`package is not outdated: ${name}`);
+            } else {
+              behindPatch.push(package);
+            }
+          } else {
+            behindMinor.push(package);
+          }
         } else {
-          behindPatch.push(package);
+          behindMajor.push(package);
         }
```

Comment:
might be able to flatten the deep conditional nest here, nit tho

fn flattening
``` javascript
const updateVersions = (behindMajor, behindMinor, currentPatch, package) => {
  if (currentMajor < latestMajor) {
    behindMajor.push(package)
    return
  }
  
  if (currentMinor < latestMinor ) {
    behindMinor.push(package)
    return
  }
  
  if (currentPatch < latestPatch) {
    behindPatch.push(package)
    return
  }
  
  throw new Error('package ...');
}
```

dirty inline flattening
```javascript
const major = [currentMajor, latestMajor, behindMajor]
const minor = [currentMinor, latestMinor, behindMinor]
const patch = [currentPatch, latestPatch, behindPatch]

const version = [major, minor, patch]
for (let i = 0; i < version.length; i++) {
  const [current, latest, behind] = version[0]
  if (current < latest) {
    behind.push(package)
    break
  }

  i == version.length - 1 ? throw new Error('package ....') : null
}
```

Reply:
Nice I will try it out.